### ISSUE TITLE
-Dkotest.tags.include= only takes into account @Tags

### ISSUE THREAD

**Issue by sksamuel** [kotest/kotest#1536]
With the recent change to @Tags, the absence of a Tag causes a spec to be skipped, but we should instantiate it and check the tags declared inside it.
Any beforeSpec callbacks should be deferred until the class is created and evaluated.

**Comment by ryanlewis** [comment]
Stumbled into this problem today. Looks like on https://github.com/kotest/kotest/blob/master/kotest-core/src/commonMain/kotlin/io/kotest/core/extensions/TagFilteredDiscoveryExtension.kt#L16 , need to also grab all tags declared inside as well.

**Comment by ryanlewis** [comment]
```kotlin
object TagFilteredDiscoveryExtension : DiscoveryExtension {
   override fun afterScan(classes: List<KClass<out Spec>>): List<KClass<out Spec>> {
      return classes.filter { cls ->
         val tags = cls.annotation<Tags>()?.let { anno -> anno.values.map { StringTag(it) } } ?: emptyList()
         val allTags = tags + cls.createInstance().tags()
         Project.tags().isActive(allTags.toSet())
      }
   }
}
```

Seems to work, but relies on initialising the test class, so possibly some unintended side effects.

**Comment by sksamuel** [comment]
Fixed in 4.1.2
